Content.Shared/_RMC14/Xenonids/Acid/ServerCorrodingEvent.cs

```diff
@@ -0,0 +1,15 @@
+﻿using Robust.Shared.Serialization;
+
+namespace Content.Shared._RMC14.Xenonids.Acid;
+
+[Serializable, NetSerializable]
+public sealed partial class ServerCorrodingEvent : EntityEventArgs
```

Comment:
> can change the building corroding logic from a hardcoded component check to just an event handler.

Can you elaborate a bit please? Given I'm not exactly comfortable with the code base and best practices yet.

Reply:
makes it possible to have different acid logic than just buildings without changing this system, making your own that cancels the event is all you have to do.

if you dont then it would require big if else chain slop from yanderedev

Reply:
So, I still have some questions. Now I have an event that contains dps and expendable light dps. I can also cancel that event, in which case regular time-based corroding component will not be added. So canceling the event assumes event was taken care of by other means, either adding DamageableCorrodingComponent, which is then processed by the SharedAcidSystem every 10 seconds or adjusting ExpendableLight parameters.

So I now need to have 2 different server side subscribers to this event:
* ExpendableLight comp subscriber will do the same thing it currently does.
* BarricadeComponent comp subscriber will add DamageableCorrodingComponent to the target.

Does that sound right? I really want to understand what is going on rather then just copy-pasting stuff around.

Reply:
yes each different method of corroding subscribes to it, like that.

if you want the light to have dps *and* be deleted over time dont cancel the event, which i think is how it works in ss13 though could be wrong

Reply:
Makes sense. Done. Thanks.